src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -61,12 +64,13 @@ internal AuthenticationOptions BaseOptions
 
         public bool Faulted { get; set; }
 
-        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger)
+        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger, [NotNull] IUrlEncoder encoder)
         {
             _baseOptions = options;
             Context = context;
             RequestPathBase = Request.PathBase;
             Logger = logger;
+            UrlEncoder = encoder;
```

Comment:
Also Twitter uses it, so 4.  Feels kind of the logger to me, we want to encourage people to use the UrlEncoder right?  So that said, if you feel super strongly that we should keep the base handler properties as slim as possible, I can remove it.  But generally I don't really see how it hurts if we already have 4 handlers ourselves that use it...